test/integration/multikueue/jobs_test.go

```diff
@@ -264,33 +264,44 @@ var _ = ginkgo.Describe("MultiKueue", ginkgo.Ordered, ginkgo.ContinueOnFailure,
 		ginkgo.By("finishing the worker job", func() {
 			reachedPodsReason := "Reached expected number of succeeded pods"
 			finishJobReason := "Job finished successfully"
+			// Conditions that we will add to the remote job to indicate job completions,
+			// and the same conditions that we expect to see on the local job status.
+			testJobConditions := make([]batchv1.JobCondition, 2)
+
 			gomega.Eventually(func(g gomega.Gomega) {
 				createdJob := batchv1.Job{}
 				g.Expect(worker1TestCluster.client.Get(worker1TestCluster.ctx, client.ObjectKeyFromObject(job), &createdJob)).To(gomega.Succeed())
 				now := metav1.Now()
-				createdJob.Status.Conditions = append(createdJob.Status.Conditions,
-					batchv1.JobCondition{
-						Type:               batchv1.JobSuccessCriteriaMet,
-						Status:             corev1.ConditionTrue,
-						LastProbeTime:      now,
-						LastTransitionTime: now,
-						Message:            reachedPodsReason,
-					},
-					batchv1.JobCondition{
-						Type:               batchv1.JobComplete,
-						Status:             corev1.ConditionTrue,
-						LastProbeTime:      now,
-						LastTransitionTime: now,
-						Message:            finishJobReason,
-					},
-				)
+				testJobConditions[0] = batchv1.JobCondition{
+					Type:               batchv1.JobSuccessCriteriaMet,
```

Comment:
This may not be expected on older versions of k8s (before 1.32), please don't check.